tools/deb/istio.mk

```diff
@@ -0,0 +1,83 @@
+.PHONY: docker.istio.deb istio.deb sidecar.deb ${OUT}/istio-sidecar.deb
+
+# Make the deb image using the CI/CD image and docker.
+docker.istio.deb:
+	(cd ${TOP}; docker run --rm -u $(shell id -u) -it \
+        -v ${GO_TOP}:${GO_TOP} \
+        -w ${PWD} \
+        -e USER=${USER} \
+        -e GOPATH=${GOPATH} \
+		--entrypoint /usr/bin/make ${CI_HUB}/ci:${CI_VERSION} \
+		sidecar.deb )
+
+# Create the 'sidecar' deb, including envoy and istio agents and configs.
+# This target uses a locally installed 'fpm' - use 'docker.sidecar.deb' to use
+# the builder image.
+# TODO: consistent layout, possibly /opt/istio-VER/...
+sidecar.deb: ${OUT}/istio-sidecar.deb
+
+istio.deb: ${OUT}/istio-sidecar.deb
```

Comment:
Will the targets "sidecar.deb" and "istio.deb" eventually make different artifacts?  If not is there a reason to have both (or add a new one), or will only one eventually survive in the long term?  If istio.deb is the long-term name and sidecar.deb is mostly there for compatibility then it might be worthwhile to update the docker.istio.deb target (and release/cloud_builder.sh) to invoke "make istio.deb" instead of "make sidecar.deb".

Reply:
I was thinking that 0.5 goes ahead with the current name, and in 0.6 we switch to istio.deb.
(we also need to add tests and sytemctl scripts for pilot, mixer, auth servers). The idea of having a single easy to install deb that can run a local istio is not bad - but we'll need few iterations.